flagsmith-core.ts

```diff
@@ -818,15 +818,19 @@ const Flagsmith = class {
             this.evaluateFlag(key, "VALUE");
         }
 
+        if (res === null && typeof options?.fallback !== 'undefined') {
+            return options.fallback;
+        }
+
         if (options?.json) {
             try {
                 if (res === null) {
-                    this.log("Tried to parse null flag as JSON: " + key)
-                    return options.fallback;
+                    this.log("Tried to parse null flag as JSON: " + key);
+                    return undefined;
```

Comment:
I changed this to `undefined` as we can only end up with an `undefined` fallback here anyway.
Returning `options.fallback` seemed a bit confusing.

Reply:
That makes sense, though I think returning null might be slightly less confusing right ?

Reply:
I just tried to match previous behavior which would have been `undefined` as it simply returned `options.fallback`.